projects/hipsparse/CMakeLists.txt

```
@@ -1,289 +1,278 @@
-    rocm_set_os_id(CLIENTS_OS)
-    string(TOLOWER "${CLIENTS_OS}" CLIENTS_OS)
-    rocm_read_os_release(CLIENTS_OS_VERSION VERSION_ID)
-  endif()
-  message(STATUS "OS: ${CLIENTS_OS} ${CLIENTS_OS_VERSION}")
-  set(GFORTRAN_RPM "libgfortran4")
-  set(GFORTRAN_DEB "libgfortran4")
-  find_package(OpenMP QUIET)
-  if(OPENMP_FOUND)
-    set(OPENMP_RPM "libgomp")
-    set(OPENMP_DEB "libomp-dev")
-  endif()
-  if(CLIENTS_OS STREQUAL "centos" OR CLIENTS_OS STREQUAL "rhel" OR CLIENTS_OS STREQUAL "almalinux")
-    if(CLIENTS_OS_VERSION VERSION_GREATER_EQUAL "8")
-      set(GFORTRAN_RPM "libgfortran")
+if(HIPSPARSE_ENABLE_CLIENT OR HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+    if(HIPSPARSE_ENABLE_OPENMP)
+        find_package(OpenMP REQUIRED)
     endif()
-  elseif(CLIENTS_OS STREQUAL "ubuntu" AND CLIENTS_OS_VERSION VERSION_GREATER_EQUAL "20.04")
-    set(GFORTRAN_DEB "libgfortran5")
-  elseif(CLIENTS_OS STREQUAL "sles" AND OPENMP_FOUND)
-    set(OPENMP_RPM "libgomp1")
-  elseif(CLIENTS_OS STREQUAL "mariner" OR CLIENTS_OS STREQUAL "azurelinux")
-    set(GFORTRAN_RPM "gfortran")
-  endif()
-
-  rocm_package_setup_component(clients)
-  rocm_package_setup_client_component(clients-common)
-  if(BUILD_CLIENTS_TESTS)
-    rocm_package_setup_client_component(
-      tests
-      DEPENDS
-        COMPONENT clients-common
-        DEB "${GFORTRAN_DEB}"
-        RPM "${GFORTRAN_RPM}")
-    rocm_package_setup_client_component(
-      tests
-      DEPENDS
-        COMPONENT clients-common
-        DEB "${OPENMP_DEB}"
-        RPM "${OPENMP_RPM}"
-    )
-  endif()
-
-  if( BUILD_CLIENTS_BENCHMARKS )
-  rocm_package_setup_client_component(
-    benchmarks
-    DEPENDS
-      COMPONENT clients-common
-      DEB "${OPENMP_DEB}"
-      RPM "${OPENMP_RPM}"
-  )
-  endif()
-
-  add_subdirectory( clients )
-
-  if(NOT WIN32)
-    rocm_package_add_rpm_dependencies(COMPONENT tests DEPENDS "${GFORTRAN_RPM}")
-    rocm_package_add_deb_dependencies(COMPONENT tests DEPENDS "${GFORTRAN_DEB}")
-    rocm_package_add_rpm_dependencies(COMPONENT benchmarks DEPENDS "${GFORTRAN_RPM}")
-    rocm_package_add_deb_dependencies(COMPONENT benchmarks DEPENDS "${GFORTRAN_DEB}")
-  endif()
-  if(TARGET OpenMP::OpenMP_CXX)
-    rocm_package_add_rpm_dependencies(COMPONENT tests DEPENDS "${OPENMP_RPM}")
-    rocm_package_add_deb_dependencies(COMPONENT tests DEPENDS "${OPENMP_DEB}")
-  endif()
 endif()
 
-# Package specific CPACK vars
-if(NOT USE_CUDA)
-  set(ROCSPARSE_MINIMUM "4.0.1")
-  rocm_package_add_dependencies(SHARED_DEPENDS "rocsparse >= ${ROCSPARSE_MINIMUM}")
-  rocm_package_add_deb_dependencies(STATIC_DEPENDS "rocsparse-static-dev >= ${ROCSPARSE_MINIMUM}")
-  rocm_package_add_rpm_dependencies(STATIC_DEPENDS "rocsparse-static-devel >= ${ROCSPARSE_MINIMUM}")
+# --- Library target ---
+
+if(NOT HIPSPARSE_BUILD_CLIENTS_ONLY)
+    add_subdirectory(library)
+else()
+    # When building clients-only, find the installed hipSPARSE library
+    find_package(hipsparse 4.0.0 REQUIRED CONFIG)
 endif()
```

Negation hurts the brain:

```suggestion
if(HIPSPARSE_ENABLE_HOST)
    add_subdirectory(library)
endif()
```
Then add the following to the clients dir with the correct logic:

```
find_package(hipsparse REQUIRED)
```

We can probably remove that option.